src/tabpfn_extensions/unsupervised/unsupervised.py

```diff
@@ -346,7 +346,7 @@ def impute_(
             else:
                 # Convert numpy arrays to tensors if necessary before stacking
                 tensor_densities = [
-                    torch.tensor(d) if isinstance(d, np.ndarray) else d
+                    d if torch.is_tensor(d) else torch.as_tensor(d, dtype=torch.float32)
```

Comment:
```suggestion
                    torch.as_tensor(d, dtype=torch.float32)
```
torch.as_tensor returns the identity if d is a tensor already and dtype matches.